File: fetch2/src/main/java/com/tonyodev/fetch2/DownloadListener.java
package com.tonyodev.fetch2;


import android.support.annotation.NonNull;

interface DownloadListener {
    void onComplete(long id, int progress, long downloadedBytes, long totalBytes);
    void onError(long id, @NonNull Error error, int progress, long downloadedBytes, long totalBytes);
    void onProgress(long id, int progress, long downloadedBytes, long totalBytes);
    void onPause(long id, int progress, long downloadedBytes, long totalBytes);
    void onCancelled(long id, int progress, long downloadedBytes, long totalBytes);
    void onRemoved(long id, int progress, long downloadedBytes, long totalBytes);
}

File: fetch2/src/main/java/com/tonyodev/fetch2/DownloadListener.kt
package com.tonyodev.fetch2

interface DownloadListener {
    fun onComplete(id: Long, progress: Int, downloadedBytes: Long, totalBytes: Long)
    fun onError(id: Long, error: Error, progress: Int, downloadedBytes: Long, totalBytes: Long)
    fun onProgress(id: Long, progress: Int, downloadedBytes: Long, totalBytes: Long)
    fun onPause(id: Long, progress: Int, downloadedBytes: Long, totalBytes: Long)
    fun onCancelled(id: Long, progress: Int, downloadedBytes: Long, totalBytes: Long)
    fun onRemoved(id: Long, progress: Int, downloadedBytes: Long, totalBytes: Long)
}
